### PR TITLE
[REF] web,*: router service is no longer a service

### PR DESCRIPTION
**Pull request by jpp-odoo**
This commit aims at simplifying the way developers can access
router related information in the webclient. Before this commit,
in components, they had to import the hook `useService` and in
the setup do something like `this.router = useService("router")`. In
services, they had to explicitly declare the dependency to the
router service. In any other functions (like helper functions),
it was a bit painful: the function had to take an env (or the
needed information directly) in arguments, which wasn't ideal.

This commit makes accessing router information easier. The router
service has been removed. Instead, we introduce a `router` object
which can be imported directly from everywhere. Something
similar has already been done for the rpc and orm services.

Part of task 3621046

### CODE REVIEW

File: addons/mail/static/src/core/common/messaging_service.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import { _t } from "@web/core/l10n/translation";
 import { rpc } from "@web/core/network/rpc";
 import { registry } from "@web/core/registry";
 import { Deferred } from "@web/core/utils/concurrency";
+import { router } from "@web/core/browser/router";
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/messaging_service.js:9]: (minor) I would sort them alphabetically by path

File: addons/mail/static/src/core/common/thread_service.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { rpc } from "@web/core/network/rpc";
 import { registry } from "@web/core/registry";
 import { memoize } from "@web/core/utils/functions";
 import { escape } from "@web/core/utils/strings";
+import { router } from "@web/core/browser/router";
```
Comment (seb-odoo) [inline addons/mail/static/src/core/common/thread_service.js:13]: (minor) I would sort them alphabetically by path

**Review by seb-odoo — APPROVED** [review]

**Review by pro-odoo — APPROVED** [review]

File: addons/web/static/tests/webclient/actions/load_state_tests.js [hunk 3]
```diff
@@ -71,6 +72,7 @@ QUnit.module("ActionManager", (hooks) => {
     QUnit.test("action and menu loading", async (assert) => {
         assert.expect(3);
         const webClient = await createWebClient({ serverData });
+        patchWithCleanup(router, makeFakeRouter());
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/actions/load_state_tests.js:75]: why is this necessary? We already patch router in makeTestEnv

**Review by aab-odoo — APPROVED** [review]
nice PR

**Review by aab-odoo — APPROVED** [review]
robodoo r+
